validphys2/src/validphys/photon/structure_functions.py

```diff
@@ -0,0 +1,113 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+import pineappl
+from scipy.interpolate import RectBivariateSpline
+
+
+class StructureFunction(ABC):
+    """Abstract class for the DIS structure functions"""
+
+    @abstractmethod
+    def fxq(self, x, Q):
+        pass
+
+
+class InterpStructureFunction(StructureFunction):
+    """
+    Compute an interpolated structure function convoluting an FKtable
+    with a PDF.
+    """
+
+    def __init__(self, path_to_fktable, pdfs):
+        self.fktable = pineappl.fk_table.FkTable.read(path_to_fktable)
+        self.pdfs = pdfs
+        self.pdgid = int(pdfs.set().get_entry("Particle"))
+        self.produce_interpolator()
+
+    def produce_interpolator(self):
+        """Produce the interpolation function to be called in fxq."""
+        x = np.unique(self.fktable.bin_left(1))
+        q2 = np.unique(self.fktable.bin_left(0))
+        self.xmin = min(x)
+        self.qmin = min(np.sqrt(q2))
+        self.qmax = max(np.sqrt(q2))
+
+        predictions = self.fktable.convolute_with_one(self.pdgid, self.pdfs.xfxQ2)
+        # here we require that the (x,Q2) couples that we passed
+        # to pinefarm is a rectangular matrix
```

Comment:
```suggestion
```

It's pretty clear what `reshape` does. But when is this passed to pinefarm? Isn't the structure function interpolator output passed to fiatlux?